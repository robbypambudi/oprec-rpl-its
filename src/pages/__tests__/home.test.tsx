import '@testing-library/jest-dom';
import '@tanstack/react-query';

import { act, render, screen } from '@testing-library/react';

import Home from '@/pages/index.page';

jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn(() => ({
    data: {
      data: [
        {
          id_: 1,
          country: 'Indonesia',
          role: 'Frontend Developer',
          experience: '1-2 years',
          developer: 'Junior',
        },
      ],
    },
  })),
}));

describe('Home Page', () => {
  it('renders correctly', () => {
    render(<Home />);

    // Home
    expect(screen.queryAllByText('Home')[0]).toBeVisible();
    // snapshoot
    expect(screen.queryAllByText('Home')[0]).toMatchSnapshot();
  });
  //
  // sentry
  // Get Title
  it('Get Title', () => {
    render(<Home />);
    expect(screen.getAllByRole('heading', { level: 1 })[0]).toBeInTheDocument();
    expect(screen.getAllByRole('heading', { level: 1 })[0]).toBeVisible();
    expect(screen.getAllByRole('heading', { level: 1 })[0]).toMatchSnapshot();
  });

  it('Get Agency', () => {
    act(() => {
      render(<Home />);
    });

    expect(screen.getAllByText('Agencies')[0]).toBeInTheDocument();
    expect(screen.getAllByText('Agencies')[0]).toBeVisible();
    expect(screen.getAllByText('1')[0]).toBeInTheDocument();
    expect(screen.getAllByText('1')[0]).toBeVisible();
  });
});
