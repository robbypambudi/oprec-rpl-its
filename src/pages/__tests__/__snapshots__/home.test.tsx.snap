// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Get Title 1`] = `
<h1
  class="text-4xl font-bold md:text-5xl text-black"
>
  Easy to 
  <span
    class="text-primary-500"
  >
    Hire
  </span>
</h1>
`;

exports[`Home Page renders correctly 1`] = `
<a
  class="hover:text-gray-600"
  href="#home"
>
  Home
</a>
`;
